java/gradle.java/src/org/netbeans/modules/gradle/java/queries/GradleCompilerOptionsQuery.java

```diff
@@ -49,9 +49,8 @@ public final class GradleCompilerOptionsQuery implements CompilerOptionsQueryImp
 
     public GradleCompilerOptionsQuery(Project project) {
         this.project = project;
-        final NbGradleProject watcher = NbGradleProject.get(project);
```

Comment:
Consider leaving the `watcher` instance and its usage here. The listener is attached to a certain NbGradleProject instance and receives events from it. With the change, it will check the **current** NbGradleProject instance which may be different from the one that actually fires the event. But maybe it's intentional ?

Reply:
The project here is final and the NbGradleProjectImpl has a 1 to 1 assignment of it's NbGradleProject on it's constructor. So I think we are safe here.